Source/src/Importers/MeshImporter.cpp

```diff
@@ -5,133 +5,131 @@
+    r_mesh->tex_coords = new float[r_mesh->buffer_sizes[static_cast<int>(ResourceMesh::Buffers::TEX_COORDS)]];
+    for (unsigned i = 0; i < ai_mesh->mNumVertices; i++)
+    {
+        memcpy(&r_mesh->tex_coords[i * 2], &ai_mesh->mTextureCoords[0][i], 2 * sizeof(unsigned));
+    }
+
+    r_mesh->buffer_sizes[static_cast<int>(ResourceMesh::Buffers::INDICES)] = ai_mesh->mNumFaces * 3;
+    r_mesh->indices = new unsigned[r_mesh->buffer_sizes[static_cast<int>(ResourceMesh::Buffers::INDICES)]];
+    for (unsigned i = 0; i < ai_mesh->mNumFaces; ++i)
+    {
+        memcpy(&r_mesh->indices[i * 3], ai_mesh->mFaces[i].mIndices, 3 * sizeof(unsigned));
+    }
+
+    r_mesh->GenerateBuffers();
+    r_mesh->GenerateAABB();
+    r_mesh->loaded = true;
+
+    // Save resource
+
+    return r_mesh;
 }
 
-void MeshImporter::Save(ResourceMesh* r_mesh, Hachiko::UID uid)
-{	
-	std::string file_path = std::string(LIBRARY_MESH_FOLDER "/") + std::to_string(uid) + ".mesh";
-
-	unsigned* sizes = r_mesh->buffer_sizes;
-	unsigned header[ResourceMesh::Buffers::n_buffers] = {
-		sizes[ResourceMesh::Buffers::b_indices],
-		sizes[ResourceMesh::Buffers::b_vertices],
-		sizes[ResourceMesh::Buffers::b_normals],
-		sizes[ResourceMesh::Buffers::b_tex_coords],
-	};
-
-	unsigned file_size = 0;
-	file_size += sizeof(header);
-	file_size += sizeof(unsigned) * sizes[ResourceMesh::Buffers::b_indices];
-	file_size += sizeof(float) * sizes[ResourceMesh::Buffers::b_vertices];
-	file_size += sizeof(float) * sizes[ResourceMesh::Buffers::b_normals];
-	file_size += sizeof(float) * sizes[ResourceMesh::Buffers::b_tex_coords];
-	
-	char* file_buffer = new char[file_size];
-	char* cursor = file_buffer;
-	unsigned size_bytes = 0;
-
-	size_bytes = sizeof(header);
-	memcpy(cursor, sizes, size_bytes);
-	cursor += size_bytes;
-
-	size_bytes = sizeof(unsigned) * sizes[ResourceMesh::Buffers::b_indices];
-	memcpy(cursor, r_mesh->indices, size_bytes);
-	cursor += size_bytes;
-
-	size_bytes = sizeof(float) * sizes[ResourceMesh::Buffers::b_vertices];
-	memcpy(cursor, r_mesh->vertices, size_bytes);
-	cursor += size_bytes;
-	
-	size_bytes = sizeof(float) * sizes[ResourceMesh::Buffers::b_normals];
-	memcpy(cursor, r_mesh->normals, size_bytes);
-	cursor += size_bytes;
-
-	size_bytes = sizeof(float) * sizes[ResourceMesh::Buffers::b_tex_coords];
-	memcpy(cursor, r_mesh->tex_coords, size_bytes);
-	cursor += size_bytes;
-
-
-	App->file_sys->Save(file_path.c_str(), file_buffer, file_size);
-	delete[] file_buffer;
+void Hachiko::MeshImporter::Save(ResourceMesh* r_mesh, UID uid)
+{
+    const std::string file_path = std::string(LIBRARY_MESH_FOLDER "/") + std::to_string(uid) + ".mesh";
+
+    const unsigned* sizes = r_mesh->buffer_sizes;
+    unsigned header[static_cast<int>(ResourceMesh::Buffers::COUNT)] = {sizes[static_cast<int>(ResourceMesh::Buffers::INDICES)], sizes[static_cast<int>(ResourceMesh::Buffers::VERTICES)],
```

Comment:
Can we split this in a new line?

Reply:
we could, of course, but with current monitors, a line that size should be a problem to read